ideascube/mediacenter/views.py

```diff
@@ -47,9 +50,17 @@ def _set_available_kinds(self, context):
             (kind, label) for kind, label in Document.KIND_CHOICES
             if kind in available_kinds]
 
+    def _set_available_package_ids(self, context):
+        available_package_ids = self._search_for_attr_from_context('package_id', context)
+        catalog = catalog_mod.Catalog()
+
+        context['available_package_ids'] = {
+            package.id:package.name for package in catalog.list_installed(available_package_ids)}
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         self._set_available_kinds(context)
+        self._set_available_package_ids(context)
         self._set_available_langs(context)
         self._set_available_tags(context)
```

Comment:
Not your fault, but this pattern it pretty ugly. Passing a mutable to a function that modifies it is fine in C where it's often the only sane way to do something, but it's really not pythonic.

Plus, modern Django (since 1.5) passes the `view` instance to the template, so the above code is the perfect example of when it would be better to simply use:

```python
    @property
    def available_kinds(self):
        ...

    @property
    def available_package_ids(self):
        ...

    ...
```

And then in the template just use `{{ view.available_kinds }}`, etc...

Again, that code predates your change (and probably also predates django 1.5 :smile: ) so I'm not sure you should move to the new style in the middle of code doing the old style. I'm just mentioning it because few people seem to know about this, and to try and sway us gently towards the new style in the future.